libs/oidc/admin/ngx/src/lib/modules/roles/edit-role/detail-role/detail-role.component.ts

```diff
@@ -34,15 +34,23 @@ export class DetailRoleComponent implements OnInit {
   public ngOnInit() {
     if (this.route.snapshot.params.roleGuid) {
       this.roleGuid = this.route.snapshot.params.roleGuid;
-      this.roleService.getRole(this.roleGuid).subscribe((role) => {
-        this.role = role;
-        this.form.patchValue(this.role);
-        this.form.valueChanges.pipe(debounceTime(1000)).subscribe(() => {
-          this.roleService
-            .updateRole(this.form.getRawValue())
-            .subscribe(() => [this.notificationService.preset('edit_role')]);
+
+      this.roleService
+        .getRole(this.roleGuid)
+        .pipe(
+          switchMap((role) => {
+            this.role = role;
+            this.form.patchValue(this.role);
```

Comment:
Small detail: Why are you assigning `this.role` to the result of a subscription (line 42)?

On a closer look `this.role` isn't being used anywhere in the template so this is just scope pollution (i.e noise). 

If it ***were*** used in the template, the thing to do would be to set the entire stream beginning n line 38 to `this.role`, and then if you wanted the value of a scope you would either subscribe to that in the template using the async pipe or if you wanted the role in another stream you could use the `withLatestFrom` operator.


```js

// Assign a stream instead of assigning its value to the result of a subscription. More declarative.
this.role = this.roleService.getRole(this.roleGuid);

this.role.pipe(
 tap((role) => {
    this.form.patchValue(role) // patchValue is a side-effect and should semantically not be part of a `switchMap` operator.
  }),
  switchMap((role) => {
    return  this.form.valueChanges.pipe(debounceTime(1000));
  })
).subscribe(() => {
  // toast 
}}

```

Reply:
Not sure exactly why it existed. Removed.